tree/dataframe/test/dataframe_datasetspec.cxx

```diff
@@ -0,0 +1,229 @@
+#include <gtest/gtest.h>
+#include <ROOT/RDataFrame.hxx>
+#include <ROOT/TestSupport.hxx>
+#include <TSystem.h>
+
+#include <thread> // std::thread::hardware_concurrency
+
+using namespace ROOT;
+using namespace ROOT::RDF;
+
+TEST(RDFDatasetSpec, SingleFileSingleColConstructor)
+{
+   auto dfWriter = RDataFrame(5).Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"});
+   dfWriter.Snapshot<int>("tree", "file.root", {"x"});
+
+   // "reference" values
+   const auto dfSimple = RDataFrame("tree", "file.root").Display<int>({"x"})->AsString();
+   const auto dfRange0 = RDataFrame("tree", "file.root").Range(2, 4).Display<int>({"x"})->AsString();
+
+   // specify only tree and file names
+   const auto dfRDS0 = RDataFrame(RDatasetSpec("tree", "file.root")).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS0, dfSimple);
+
+   // specify tree name, file name, starting index => [2, 0) => range is disregarded (all entries are displayed)
```

Comment:
This is not intented, the range of entries should either be specified or not

Reply:
Just to make sure, is the remark towards 1.`RDatasetSpec("tree", "file.root")` or 2.`RDatasetSpec("tree", "file.root", 2)`.

In either case, I have a reason to test that:
* If it is 1 => as you said the range could be specified or not, and here it is not specified
* If it is 2 => maybe a user tries to use this feature similarly to applying `Range(2)`, which is not allowed. In addition, I am testing what would happend if 2 valid indeces are entered, but the starting index is bigger than the ending one.

Reply:
Yeah it is for 2